File: app/src/main/java/com/example/flashcards/MainActivity.java
package com.example.flashcards;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;               // 1. added
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;
import com.example.flashcards.model.*;

public class MainActivity extends AppCompatActivity {

    private SubjectAdapter adapter;
    private RecyclerView grid;
    private TextView emptyTxt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        grid     = findViewById(R.id.subjectGrid);
        emptyTxt = findViewById(R.id.emptyTxt);
        ImageButton addBtn = findViewById(R.id.imageButton);

        addBtn.setOnClickListener(v ->
                startActivity(new Intent(MainActivity.this, AddCardActivity.class)));

        adapter = new SubjectAdapter(Library.getInstance().getSubjects(), this::openSubject);
        grid.setAdapter(adapter);

        refreshEmptyState();
    }

    private void openSubject(Subject subject) {
        Intent i = new Intent(this, FlashActivity.class);
        i.putExtra("SUBJECT_TITLE", subject.getTitle());
        startActivity(i);
    }

    @Override
    protected void onResume() {
        super.onResume();
        adapter.notifyDataSetChanged();
        refreshEmptyState();
    }

    /* package-visible so SubjectAdapter can call it after delete / pin */
    void refreshEmptyState() {
        boolean has = !Library.getInstance().getSubjects().isEmpty();
        grid.setVisibility(has ? View.VISIBLE : View.GONE);
        emptyTxt.setVisibility(has ? View.GONE : View.VISIBLE);
    }
}

File: app/src/main/java/com/example/cleverty/MainActivity.kt
package com.example.cleverty

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.OnApplyWindowInsetsListener
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    private var loginbtn: Button? = null
    private var signupbtn: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        loginbtn = findViewById<Button?>(R.id.login_btn)
        loginbtn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                val intent = Intent(this@MainActivity, LoginPage::class.java)
                startActivity(intent)
            }
        })

        signupbtn = findViewById<Button?>(R.id.signup_btn)
        signupbtn!!.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                val intent = Intent(this@MainActivity, SignupPage::class.java)
                startActivity(intent)
            }
        })

        ViewCompat.setOnApplyWindowInsetsListener(
            findViewById<View?>(R.id.main),
            OnApplyWindowInsetsListener { v: View?, insets: WindowInsetsCompat? ->
                val systemBars = insets!!.getInsets(WindowInsetsCompat.Type.systemBars())
                v!!.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
                insets
            })
    }

}